File: Objective-C/LRI/NetworkLayer/Endpoint/DrinkEndpoint.m
//
//  DrinkEndpoint.m
//  LRI
//
//  Created by Development on 05.07.2022.
//

#import <Foundation/Foundation.h>

#import "DrinkEndpoint.h"

@interface DrinkEndpoint ()
@end

@implementation DrinkEndpoint

- (NSString *)scheme {
    return @"https";
}

- (NSString *)method {
    return @"GET";
}

- (NSString *)baseURL {
    return @"www.thecocktaildb.com";
}

- (NSString *)path {
    return @"/api/json/v1/1/filter.php";
}

- (NSMutableArray<NSURLQueryItem *> *)parameters {
    return [[NSMutableArray alloc] initWithObjects:[[NSURLQueryItem alloc] initWithName:@"a" value:@"Alcoholic"], nil];
}

@end

File: SwiftUIKit/LRI/NetworkLayer/Endpoint/DrinkEndpoint.swift
//
//  DrinkEndpoint.swift
//  LRI
//
//  Created by Development on 01.07.2022.
//

import Foundation

enum DrinkEndpoint: Endpoint {
    case getDrinks
    
    var scheme: String {
        switch self {
        default :
            return "https"
        }
    }
    
    var method: String {
        switch self {
        case .getDrinks:
            return "GET"
        }
    }
    
    var baseURL: String {
        switch self {
        default :
            return "www.thecocktaildb.com"
        }
    }
    
    var path: String {
        switch self {
        case .getDrinks:
            return "/api/json/v1/1/filter.php"
        }
    }
    
    var parameters: [URLQueryItem]? {
        switch self {
        case .getDrinks:
            return [URLQueryItem(name: "a", value: "Alcoholic")]
        }
    }
}
